lib/action_subscriber/threadpool.rb

```diff
@@ -4,17 +4,29 @@ class Threadpool
     # Class Methods
     #
     def self.busy?
-      (pool.pool_size == pool.busy_size)
+      pools.any? do |_pool_name, pool|
+        pool.pool_size == pool.busy_size
+      end
     end
 
-    def self.perform_async(*args)
-      self.pool.async.perform(*args)
+    def self.new_pool(name, pool_size = nil)
+      fail ArgumentError, "#{name} already exists as a threadpool" if pools.key?(name)
+      pool_size ||= ::ActionSubscriber.config.threadpool_size
+      pools[name] = ::Lifeguard::InfiniteThreadpool.new(
+        :pool_size => pool_size
+      )
     end
 
-    def self.pool
-      @pool ||= ::Lifeguard::InfiniteThreadpool.new(
-        :pool_size => ::ActionSubscriber.config.threadpool_size
-      )
+    def self.pool(which_pool = :default)
+      pools[which_pool]
+    end
+
+    def self.pools
+      @pools ||= {
+        :default => ::Lifeguard::InfiniteThreadpool.new(
```

Comment:
I tried rewriting this a few different ways (including your suggestion of calling `self.add_pool(:default)`, but I can't find anything I like.

The problem with calling `self.new_pool(:default)` as I'm defining this class is that the application calling my code has not yet had a chance to call `ActionSubscriber.config` to set the default pool size. Normally we don't initialize the pool until it first gets accessed which is after the user has done their configuration.

The duplication does bother me a little, but I haven't been able to find a better way so far. Usually when I find myself in this situation I fall back to:

> Duplication is far cheaper than the wrong abstraction
> ~ Sandi Metz